### PR TITLE
Adding support of -1 as size to be passed into get and getFile

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: tests/lib/avatartest.php [hunk 1]
```diff
@@ -47,6 +45,21 @@ public function testGetAvatarSizeMatch() {
 		$this->assertEquals($expected->data(), $this->avatar->get(128)->data());
 	}
 
+	public function testGetAvatarSizeMinusZero() {
```
Comment (MorrisJobke) [inline tests/lib/avatartest.php]: MinusZero? :wink:

File: lib/public/iavatar.php [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
 	/**
 	 * get the users avatar
-	 * @param int $size size in px of the avatar, avatars are square, defaults to 64
+	 * @param int $size size in px of the avatar, avatars are square, defaults to 64, -1 can be used to not scale the image
 	 * @return boolean|\OCP\IImage containing the avatar or false if there's no image
 	 * @since 6.0.0
```
Comment (MorrisJobke) [inline lib/public/iavatar.php]: Could this be changed to:

```
@since 6.0.0 - size of -1 was added in 9.0.0
```

**Comment by nickvergessen** [comment]
```
OCA\Files_sharing\Tests\EtagPropagation::testOwnerWritesToSingleFileShare
Failed asserting that the etag for "" of user test-share-user2 has changed
Failed asserting that '5fe0cc341a6f7eb996ea61c20678271c' is not equal to <string:5fe0cc341a6f7eb996ea61c20678271c>.
```
.....

File: tests/lib/avatartest.php (hunk 1 above)
Comment (DeepDiver1975) [inline tests/lib/avatartest.php]: never code when you are ill ...

File: lib/private/avatar.php [hunk 3]
```diff
@@ -135,16 +133,16 @@ public function remove () {
...
-	 * @param int $size
-	 * @return File
-	 * @throws NotFoundException
+	 * @inheritdoc
 	 */
 	public function getFile($size) {
 		$ext = $this->getExtention();
 
-		$path = 'avatar.' . $size . '.' . $ext;
+		if ($size < 0) {
```
Comment (rullzer) [inline lib/private/avatar.php]: `$size === -1` ?

File: lib/private/avatar.php [hunk 4]
```diff
@@ -157,7 +155,9 @@ public function getFile($size) {
 			/** @var File $file */
 			$file = $this->folder->get('avatar.' . $ext);
 			$avatar->loadFromData($file->getContent());
-			$avatar->resize($size);
+			if ($size >= 0) {
```
Comment (rullzer) [inline lib/private/avatar.php]: `> 0` ?
Not really important as the above if already catches <= 0. But resizing to 0x0 does not really mean anything.

**Comment by rullzer** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
@MorrisJobke @LukasReschke mind taking a look - THX

**Comment by LukasReschke** [comment]
:+1:

**Pull request merged**